modules/internal/ChapelBase.chpl

```diff
@@ -441,7 +441,7 @@ module ChapelBase {
   //
   // bitwise operations on primitive types
   //
-  inline proc ~(a: bool) return __primitive("u~", a);
+  inline proc ~(a: bool) { compilerError("~ is not suported on bool"); }
```

Comment:
I can't make it "compiler generated" because if I do, the following line's `~` overload will apply. (I.e. `~` on `bool` will coerce the argument to int, then run bit complement). Adding an unnecessary `where` clause covers much the same territory, so I'm doing that. I've also added a test that checks that user-defined overload is possible.